py/dml/expr.py

```diff
@@ -243,6 +312,31 @@ def read(self):
         return (self.fun.read() +
                 '(' + ", ".join(e.read() for e in self.args) + ')')
 
+def mkApplyInits(site, fun, inits, location, scope):
+    '''Apply a C function to initializers'''
+    funtype = fun.ctype()
+
+    if not funtype:
+        raise EAPPLY(fun)
+
+    try:
+        funtype = realtype(funtype)
+        if isinstance(funtype, TPtr) and isinstance(funtype.base, TFunction):
+            # Pointers to functions are the same as the functions
+            funtype = realtype(funtype.base)
+    except DMLUnknownType:
+        raise ETYPE(site, funtype)
+
+    if not isinstance(funtype, TFunction):
+        raise EAPPLY(fun)
+
+    args = typecheck_inarg_inits(
+        site, inits,
+        [(str(i + 1), t) for (i, t) in enumerate(funtype.input_types)],
+        location, scope, 'function', funtype.varargs)
+
+    return Apply(site, fun, args, funtype)
+
 def mkApply(site, fun, args):
```

Comment:
IIRC, mkApply belongs to expr.py only because of `Expression.apply`; perhaps it can be demoted to ctree now?

Reply:
True, it's only in expr.py due to `Expression.apply` -- but how do you figure it could be demoted? By adding
```C
from .ctree import mkApply`
```
in `Expression.apply`?